app/js/ImageOptimiser.js

```diff
@@ -1,44 +1,41 @@
-/* eslint-disable
-    no-unused-vars,
-*/
-// TODO: This file was created by bulk-decaffeinate.
-// Fix any style issues and re-enable lint.
-/*
- * decaffeinate suggestions:
- * DS102: Remove unnecessary code created because of implicit returns
- * DS207: Consider shorter variations of null checks
- * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
- */
-const { exec } = require('child_process')
 const logger = require('logger-sharelatex')
-const Settings = require('settings-sharelatex')
+const metrics = require('metrics-sharelatex')
+const { callbackify } = require('util')
+const safeExec = require('./SafeExec').promises
 
 module.exports = {
-  compressPng(localPath, callback) {
-    const startTime = new Date()
-    logger.log({ localPath }, 'optimising png path')
-    const args = `optipng ${localPath}`
-    const opts = {
-      timeout: 30 * 1000,
-      killSignal: 'SIGKILL'
-    }
-    if (!Settings.enableConversions) {
-      const error = new Error('Image conversions are disabled')
-      return callback(error)
+  compressPng: callbackify(compressPng),
+  promises: {
+    compressPng
+  }
+}
+
+async function compressPng(localPath, callback) {
+  const timer = new metrics.Timer('compressPng')
+  logger.log({ localPath }, 'optimising png path')
+
+  const args = ['optipng', localPath]
+  const opts = {
+    timeout: 30 * 1000,
+    killSignal: 'SIGKILL'
+  }
+
+  try {
+    await safeExec(args, opts)
```

Comment:
this will return an object with `stdout` and `stderr`, you're just choosing not to collect those values, right?

Reply:
That's correct - don't need them here. (Actually, it doesn't look like we use `stdout` anywhere, but it may come in handy.)